src/Sylius/Bundle/SettingsBundle/Resolver/DefaultSettingsResolver.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\SettingsBundle\Resolver;
+
+use Doctrine\ORM\NonUniqueResultException;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * Default implementation that resolves settings in the simplest way possible. It tries to find a schema by it's name
+ * and optionally the given namespace.
+ *
+ * @author Steffen Brem <steffenbrem@gmail.com>
+ */
+class DefaultSettingsResolver implements SettingsResolverInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    protected $settingRepository;
+
+    /**
+     * @param RepositoryInterface $settingRepository
+     */
+    public function __construct(RepositoryInterface $settingRepository)
+    {
+        $this->settingRepository = $settingRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($schemaAlias, $namespace = null)
+    {
+        try {
+            $criteria = [
+                'schemaAlias' => $schemaAlias,
+            ];
+
+            if (null !== $namespace) {
+                $criteria['namespace'] = $namespace;
+            }
+
+            return $this->settingRepository->findOneBy($criteria);
+        } catch (NonUniqueResultException $e) {
```

Comment:
Well I updated my comment a bit about the unique constraint on the `schema_alias,namespace` combination. Since when you would use theme settings, you most likely only have `schema_alias` defined (and leave namespace on `null`) and then make a one-directional many-to-one (basically a many-to-many in doctrine with a unique constraint) with a `Theme` or whatever relationship you want to have with settings.

So in short, I will remove the unique constraint from the model. Is that ok?